runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -992,7 +992,7 @@ private Instant minTargetAndGcTime(Instant target) {
      * user has no way to compute a good choice of time.
      */
     private void setUnderlyingTimer(Instant target) {
-      timerInternals.setTimer(namespace, timerId, target, spec.getTimeDomain());
+      timerInternals.setTimer(namespace, timerId, timerId, target, target, spec.getTimeDomain());
```

Comment:
I think you probably want to default to empty string here

Reply:
Done